components/core/src/clp_s/JsonParser.cpp

```diff
@@ -520,6 +521,317 @@ bool JsonParser::parse() {
     return true;
 }
 
+NodeType get_archive_node_type(
+        clp::ffi::SchemaTreeNode const& node,
+        std::pair<clp::ffi::SchemaTreeNode::id_t, std::optional<clp::ffi::Value>> p
+) {
+    auto const node_type = node.get_type();
+    // figure out what type the node is in archive node type
+    NodeType archiveNodeType;
+    switch (node_type) {
+        case clp::ffi::SchemaTreeNode::Type::Int:
+            archiveNodeType = NodeType::Integer;
+            break;
+        case clp::ffi::SchemaTreeNode::Type::Float:
+            archiveNodeType = NodeType::Float;
+            break;
+        case clp::ffi::SchemaTreeNode::Type::Bool:
+            archiveNodeType = NodeType::Boolean;
+            break;
+        case clp::ffi::SchemaTreeNode::Type::UnstructuredArray:
+            archiveNodeType = NodeType::UnstructuredArray;
+            break;
+        case clp::ffi::SchemaTreeNode::Type::Str:
+            // std::cerr << "In str\n";
+            if (p.second.value().is<std::string>()) {
+                // maybe special case for date string
+                archiveNodeType = NodeType::VarString;
+            } else {
+                archiveNodeType = NodeType::ClpString;
+            }
+            break;
+        case clp::ffi::SchemaTreeNode::Type::Obj:
+            // std::cerr << "In obj\n";
+            if (p.second.has_value()) {
+                if (p.second.value().is_null()) {
+                    // std::cout << "Found Null\n";
+                    archiveNodeType = NodeType::NullValue;
+                } else {
+                    archiveNodeType = NodeType::Object;
+                }
+            } else {
+                archiveNodeType = NodeType::Object;
+            }
+            break;
+        default:
+            archiveNodeType = NodeType::Unknown;
+            break;
+    }
+    return archiveNodeType;
+}
+
+//
+int JsonParser::get_archive_node_id(
+        std::map<std::tuple<int, NodeType>, int>& cache,
+        int irNodeID,
+        NodeType archiveNodeType,
+        clp::ffi::SchemaTree const& irTree
+) {
+    std::tuple<int, NodeType> key(irNodeID, archiveNodeType);
+    if (cache.find(key) != cache.end()) {
+        return cache[key];
+    }
```

Comment:
```suggestion
    auto key = std::make_tuple(ir_node_id, archive_node_type);
    auto it = cache.find(key)
    if (cache.end() != it) {
        return *it;
    }
```
Generally you should always write lookups in c++ this way. The way it was written previously performs the lookup twice, whereas this performs the lookup only once.